### PR TITLE
[2.19.x] DDF-UI-153 readonly result forms

### PR DESCRIPTION
**Pull request by cassandrabailey293**
#### What does this PR do?
Creates consistent behavior for read only result forms.
System result forms and result forms shared to a user with only read permission will now allow you to click to open them, but all sections will be read-only and unmodifiable. 

#### Who is reviewing it? 

@hayleynorton @zta6 @andrewzimmer 

#### Select relevant component teams: 

@codice/ui 

#### How should this be tested?

1. build/profile install/ load system result forms (add forms to etc/forms folder and run `forms:load` command)
2. Share a result form with your user and make it read only
3. verify that for both the system result form and the shared result form, you can open the result form and see its details, but not modify them, and no error toast messages pop up.

#### What are the relevant tickets?
Fixes: https://github.com/codice/ddf-ui/issues/153

#### Screenshots

image

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by andrewzimmer** [comment]
Should we also hide the save/cancel buttons for the read-only view? they don't seem to have much of a purpose now

**Comment by andrewzimmer** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-forms.view.js [hunk 1]
```diff
@@ -18,28 +18,16 @@ const lightboxResultInstance = require('../lightbox/result/lightbox.result.view.
...
-          type: 'error',
-        },
-        3000
-      )
-      return
-    }
-
+    this.model.set({
+      readOnly:
+        !user.canWrite(this.model) || this.model.get('createdBy') === 'system',
```
Comment (rymach) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-form/result-forms.view.js:29]: consider moving `|| this.model.get('createdBy') === 'system'` into canWrite

**Comment by zta6** [comment]
🎉 Hero Successful 🎉 
Installed system result forms
shared result form as read only
verified that system and shared result form are both read only and that no error messages appear.
LGTM

**Comment by Lambeaux** [comment]
Build now.

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Pull request merged**